process/block/metablock.go

```diff
@@ -1091,6 +1092,7 @@ func (mp *metaProcessor) CommitBlock(
 	}
 
 	mp.indexBlock(header, body, lastMetaBlock, notarizedHeadersHashes, rewardsTxs)
+	mp.saveHistoryData(headerHash, headerHandler, bodyHandler)
```

Comment:
Not sure - why isn't `indexBlock` called on a goroutine?

Reply:
good question. because maybe already inside that method are spammed goroutines ?